tests/e2e/macros/mod.rs

```diff
@@ -1,12 +1,200 @@
-use std::{path::PathBuf, sync::LazyLock};
+use std::{fmt::Display, ops::Not, path::PathBuf, sync::LazyLock};
+
+use crate::support::{
+    MockClient,
+    cursor::peek_caret,
+    cursors,
+    diagnostics::{DiagnosticAndRelatedInfo, DiagnosticsWithUrl, get_related_diagnostic_code},
+    fixture::Fixture,
+    itertools::IteratorExtension,
+    normalize::normalize_diagnostics,
+    sandbox,
+};
+use cairo_language_server::lsp::ext::ExpandMacro;
+use itertools::Itertools;
+use lsp_types::{Position, TextDocumentIdentifier, TextDocumentPositionParams};
+use serde::Serialize;
+use serde_json::json;
+
+mod fixtures;
+mod procedural;
 
 pub const SCARB_TEST_MACROS_PACKAGE_NAME: &str = "scarb_procedural_macros";
 
-#[expect(dead_code)] // Removed in the next PR.
-pub const SCARB_TEST_MACROS_PACKAGE: LazyLock<PathBuf> = LazyLock::new(|| {
+pub static SCARB_TEST_MACROS_PACKAGE: LazyLock<PathBuf> = LazyLock::new(|| {
     PathBuf::from(env!("CARGO_MANIFEST_DIR"))
         .join("tests")
         .join(SCARB_TEST_MACROS_PACKAGE_NAME)
         .canonicalize()
         .expect("should be able to obtain an absolute path to `scarb_procedural_macros`")
 });
+
+// NOTE: This procedure is implemented as a macro to delegate
+// the choice of the snapshot destination in the file system
+// and avoid a possible mess with specifying those locations manually.
+//
+/// Tests a Cairo code snippet in the context of a [`MacroTest`].
+/// Serializes the components of the obtained report to appropriate formats
+/// (TOML for expansions and JSON for diagnostics) and tests them against the saved snapshots.
+#[macro_export]
+macro_rules! test_macro_expansion_and_diagnostics {
+    ($test_setup:ident, $code_with_cursors:expr) => {
+        let report =
+            <$test_setup as $crate::macros::MacroTest>::test(::indoc::indoc!($code_with_cursors));
+        ::insta::assert_snapshot!(report);
```

Comment:
```suggestion
        let (report, ls) =
            <$test_setup as $crate::macros::MacroTest>::test(::indoc::indoc!($code_with_cursors));
        ::insta::assert_snapshot!(report);
        drop(ls)
```

if you can delay dropping of `MockClient`, you'll get autogenerated descriptions in insta snapshots